File: server/src/main/java/de/zalando/zally/apireview/OpenApiHelper.java
package de.zalando.zally.apireview;

import io.swagger.models.Swagger;
import io.swagger.parser.SwaggerParser;

public class OpenApiHelper {

    private static String API_ID_EXTENSION = "x-api-id";

    private OpenApiHelper() {}

    public static String extractApiName(String apiDefinition) {
        try {
            Swagger openApi = new SwaggerParser().readWithInfo(apiDefinition, true).getSwagger();
            if (openApi == null || openApi.getInfo() == null) {
                return null;
            }
            return openApi.getInfo().getTitle().trim();
        } catch (Exception e) {
            return null;
        }
    }

    public static String extractApiId(String apiDefinition) {
        try {
            Swagger openApi = new SwaggerParser().readWithInfo(apiDefinition, true).getSwagger();
            if (openApi == null || openApi.getInfo() == null || !openApi.getInfo().getVendorExtensions().containsKey(API_ID_EXTENSION)) {
                return null;
            }
            return (String) openApi.getInfo().getVendorExtensions().get(API_ID_EXTENSION);
        } catch (Exception e) {
            return null;
        }
    }
}

File: server/zally-server/src/main/kotlin/org/zalando/zally/apireview/OpenApiHelper.kt
package org.zalando.zally.apireview

import io.swagger.parser.SwaggerParser

object OpenApiHelper {

    private const val API_ID_EXTENSION = "x-api-id"

    fun extractApiName(apiDefinition: String): String? = try {
        val openApi = SwaggerParser().readWithInfo(apiDefinition, true).swagger
        if (openApi == null || openApi.info == null) {
            null
        } else {
            openApi.info.title.trim { it <= ' ' }
        }
    } catch (e: Exception) {
        null
    }

    fun extractApiId(apiDefinition: String): String? = try {
        val openApi = SwaggerParser().readWithInfo(apiDefinition, true).swagger
        if (openApi == null || openApi.info == null || !openApi.info.vendorExtensions.containsKey(API_ID_EXTENSION)) {
            null
        } else {
            openApi.info.vendorExtensions[API_ID_EXTENSION] as String
        }
    } catch (e: Exception) {
        null
    }
}
